lib/fluent/plugin/in_tail.rb

```diff
@@ -352,9 +358,14 @@ def detach_watcher(tw, close_io = true)
     end
 
     def detach_watcher_after_rotate_wait(tw)
-      timer_execute(:in_tail_close_watcher, @rotate_wait, repeat: false) do
+      # Call event_loop_attach is high-cost for short-live object.
+      # If this has a problem with large number of files, use @_event_loop directly.
+      checker = ->() { true }
+      detacher = ->(watcher) { event_loop_detach(watcher) }
+      timer = Fluent::PluginHelper::Timer::TimerWatcher.new(:closer, @rotate_wait, false, log, checker, detacher) {
```

Comment:
Why do you extract `timer_execute`?
The above comment does not describe the reason to extract `timer_execute`.

I think we should create another PR for extracting `timer_execute` and use `@event_loop_directly`.

BTW, how about adding `TODO:` to the above comment?

Reply:
Ah, I forgot to back to timer_execute implementation.
Fix soon.